components/datagrid/DataGrid.js

```diff
@@ -119,7 +119,8 @@ class DataGrid extends PureComponent {
       [theme['has-border-right']]: selectable || stickyFromLeft > 0,
     });
 
-    const isPartiallySelected = selectedRows.length > 0 && selectedRows.length !== children[1].length;
+    const numberOfBodyRows = children.flat().filter(child => isComponentOfType(BodyRow, child)).length;
```

Comment:
Are we polyfilling flat() ?
It's a VERY new array function.

Reply:
Upon checking, its indeed very new. Changed it so I use React.Children.toArray() to flatten it out.

 :ok: by @Kemosabert on 25/07/2018 09:52:32